sdk/python/kfp/_auth.py

```diff
@@ -190,3 +209,161 @@ def id_token_from_refresh_token(client_id, client_secret, refresh_token, audienc
+    cache = DiskCache('kfp')
+
+    if from_cache:
+        cached_creds = cache.get(expires_in=86400)
+        if cached_creds:
+            return cached_creds
+        else:
+            logging.info(
+                'Credentials are not found in the cache, trying to login')
+
+    driver = get_chrome_driver()
+
+    driver.get(host)
+
+    username, password = validate_creds(username, password)
+
+    # Setting the value of email input field
+    driver.execute_script(
+        'var element = document.getElementById("signInFormUsername");' +
+        f'element.value = "{username}";')
+
+    # Setting the value of password input field
+    driver.execute_script(
+        'var element = document.getElementById("signInFormPassword");' +
+        f'element.value = "{password}";')
+
+    # Submitting the form or click the sign in button
+    driver.execute_script(
+        'document.getElementsByName("signInSubmitButton")[0].click();')
+
+    cookies_list = driver.get_cookies()
+    auth_cookie = f"{cookies_list[0]['name']}={cookies_list[0]['value']}"
+    cache.save(auth_cookie)
+
+    return auth_cookie
+
+def get_chrome_driver(driver_index=-1):
+    """ Download Chrome driver if it doesn't already exists
+    """
+
+    options = Options()
+    options.add_argument('--headless')
+    options.add_argument('--disable-gpu')
+
+    if not os.path.exists(DEFAULT_CHROME_DRIVER_PATH):
+        logging.info('Selenium driver not found, trying to download one')
+        download_driver()
+        driver_index = 0
+
+    try:
+        return Chrome(executable_path=DEFAULT_CHROME_DRIVER_PATH,
+                      chrome_options=options)
+    except SessionNotCreatedException as e:
+        if "This version of ChromeDriver only supports Chrome version" in e.msg:
+            logging.info('Chrome driver mismatch, downloading newer version')
+            os.remove(DEFAULT_CHROME_DRIVER_PATH)
+            download_driver(driver_index=driver_index + 1)
+            return get_chrome_driver(driver_index=driver_index + 1)
+        else:
+            raise
+
+def download_driver(driver_index=0):
+    """ Helper function for downloading the driver
+    """
+
+    osy = CHROME_DRIVER_SUPPORTED_PLATFORMS[platform.system()]
+
+    try:
+        driver_url = (f'{CHROME_DRIVER_ROOT_URL}/{CHROME_DRIVER_VERSIONS[driver_index]}'
+                      f'/chromedriver_{osy}.zip')
+    except IndexError:
+        raise Exception(
+            ('Suitable chrome driver not found, please check your chrome '
+             f'version and download appropriate driver to {DEFAULT_CHROME_DRIVER_PATH}'
+            ))
+
+    logging.info('Downloading driver from %s', driver_url)
+
+    r = requests.get(driver_url)
+    z = zipfile.ZipFile(io.BytesIO(r.content))
+    path = os.path.dirname(DEFAULT_CHROME_DRIVER_PATH)
+
+    if not os.path.exists(path):
+        os.makedirs(path)
+
+    z.extractall(path)
+    os.chmod(DEFAULT_CHROME_DRIVER_PATH, stat.S_IEXEC)
+
+
+def mkdir(path):
```

Comment:
If defining this here it makes sense to use it on line 318 as well.

In both places I'm curious if there is a reason to avoid the following:
`os.makedirs(path, exist_ok=True)`